src/live.rs

```diff
@@ -695,6 +696,37 @@ fn collect_iso_records(
     Ok(())
 }
 
+pub fn iso_extract_pxe(config: &IsoExtractPxeConfig) -> Result<()> {
+    let mut iso = IsoFs::new(&config.input)?;
+    let pxeboot = match iso.get_path(COREOS_ISO_PXEBOOT_DIR)? {
+        iso9660::DirectoryRecord::Directory(d) => d,
+        iso9660::DirectoryRecord::File(_) => {
+            bail!("expected dir {}, but found file", COREOS_ISO_PXEBOOT_DIR)
+        }
+    };
+
+    let outdir = config.output_dir.as_deref().unwrap_or(".");
+    std::fs::create_dir_all(outdir)?;
+
+    for record in iso.list_dir(&pxeboot)? {
+        match record? {
+            iso9660::DirectoryRecord::Directory(_) => continue,
+            iso9660::DirectoryRecord::File(file) => {
+                let basename = file.name.trim_end_matches(";1").trim_end_matches('.');
+                let path = Path::new(outdir).join(basename);
+                let mut outf = OpenOptions::new()
+                    .write(true)
+                    .create_new(true)
+                    .open(&path)
+                    .with_context(|| format!("opening {}", path.display()))?;
+                println!("Extracting {} to {}", basename, path.display());
+                copy(&mut iso.read_file(&file)?, &mut outf)?;
```

Comment:
In my tests this is using 8 KiB chunks.  It probably wants a `BufReader`/`BufWriter`.